core/arch/arm/include/arm64_macros.S

```diff
@@ -116,6 +116,24 @@
 		movk    \_reg, :abs_g0_nc:\_val
 	.endm
 
+	.macro add_imm _reg, _val
+	.if ((\_val) > 0xfff)
+		add	\_reg, \_reg, ((\_val) >> 12), LSL #12
+	.endif
+	.if (((\_val) & 0xfff) > 0)
```

Comment:
Is `> 0` needed?

Reply:
We can potentially save one instruction.